controller/instance_manager_controller.go

```diff
@@ -1535,6 +1530,39 @@ func (imc *InstanceManagerController) createGenericManagerPodSpec(im *longhorn.I
 	return podSpec, nil
 }
 
+func (imc *InstanceManagerController) setPodExtraAnnotations(podSpec *corev1.Pod) error {
+	storageNetwork, err := imc.ds.GetSettingWithAutoFillingRO(types.SettingNameStorageNetwork)
+	if err != nil {
+		return err
+	}
+
+	nadAnnot := string(types.CNIAnnotationNetworks)
+	if storageNetwork.Value != types.CniNetworkNone {
+		podSpec.Annotations[nadAnnot] = types.CreateCniAnnotationFromSetting(storageNetwork)
+	}
+
+	// Get default backup target
+	backupTarget, err := imc.ds.GetBackupTargetRO(types.DefaultBackupTargetName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		imc.logger.WithError(err).Warnf("Failed to find the %s backup target", types.DefaultBackupTargetName)
+		return nil
```

Comment:
Can we support multiple backup targets?

Reply:
No, IIUC, we can not assign multiple IAM ARN roles to a pod.
Therefore, I chose the default backup target secret to be the IAM role secret when implementing the multiple backup targets.

If we allow users to choose the secret of a backup target or an individual secret to be the IAM role secret, we will need to add a field in the backup target CRD or a parameter in the secretCR.Data.

Reply:
Improvement ticet: https://github.com/longhorn/longhorn/issues/12170